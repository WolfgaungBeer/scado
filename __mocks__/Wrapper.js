import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
