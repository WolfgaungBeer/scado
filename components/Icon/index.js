import React from 'react';
import { string, object } from 'prop-types';
import { withTheme } from 'styled-components';

const propTypes = {
    icon: string.isRequired,
    color: string,
    scale: string,
    theme: object.isRequired
};

const defaultProps = {
    color: undefined,
    scale: undefined
};

const Icon = ({ icon, color, scale, theme }) => {

    const themeColor = theme.color[color] || 'inherit';
    const themeScale = theme.scale[scale] || 'inherit';

    const style = {
        color: themeColor,
        fontSize: themeScale,
        transform: 'translateY(15%)'
    };

    return (<i className="material-icons" style={style}>{icon}</i>);
};

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default withTheme(Icon);
