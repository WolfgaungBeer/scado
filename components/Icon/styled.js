import styled, { keyframes, css } from 'styled-components';

const rotationKeyframes = keyframes`
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
`;

const StyledIcon = styled.i`
    ${(props) => {
        if (props.loading) {
            return css`
                display: inline-block;
                animation: ${rotationKeyframes} 0.85s linear infinite;
            `;
        }
        return '';
    }};
    color: ${props => props.theme.color[props.color] || 'inherit'};
    font-size: ${props => props.theme.scale[props.scale] || 'inherit'};
`;

export default StyledIcon;
