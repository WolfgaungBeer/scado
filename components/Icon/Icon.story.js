import React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import Theme from '../Theme/Theme';
import defaultTheme from '../Theme/defaultTheme';
import Text from '../Text';
import Icon from './';

import '../../node_modules/material-design-icons/iconfont/material-icons.css';

const IconWrapper = styled.div`
    color: #FF0FF0;
`;

storiesOf('Icon', module)
    .add('default', () => (
        <Theme theme={defaultTheme}>
            <Icon icon="group_work"/>
        </Theme>
    ))
    .add('color error and size l', () => (
        <Theme theme={defaultTheme}>
            <Icon icon="3d_rotation" color="error" scale="l" />
        </Theme>
    ))
    .add('color success and size xxl', () => (
        <Theme theme={defaultTheme}>
            <Icon icon="check_circle" color="success" scale="xxl" />
        </Theme>
    ))
    .add('wrapped in a Text.H1', () => (
        <Theme theme={defaultTheme}>
            <Text.H1><Icon icon="dashboard" />Icon Test</Text.H1>
        </Theme>
    ))
    .add('wrapped in a Text.H4', () => (
        <Theme theme={defaultTheme}>
            <Text.H4><Icon icon="dashboard" />Icon Test</Text.H4>
        </Theme>
    ))
    .add('wrapped in a Text.P', () => (
        <Theme theme={defaultTheme}>
            <Text.P><Icon icon="dashboard" />Icon Test</Text.P>
        </Theme>
    ))
    .add('wrapped in a Text.Label', () => (
        <Theme theme={defaultTheme}>
            <Text.Label><Icon icon="dashboard" />Icon Test</Text.Label>
        </Theme>
    ))
    .add('wrapped in a Text.Label with color and scale', () => (
        <Theme theme={defaultTheme}>
            <Text.Label color="success" scale="xxl"><Icon icon="dashboard" />Icon Test</Text.Label>
        </Theme>
    ))
    .add('custom icon', () => (
        <Theme theme={defaultTheme}>
            <IconWrapper>
                <Icon icon="dashboard" />
            </IconWrapper>
        </Theme>
    ));
