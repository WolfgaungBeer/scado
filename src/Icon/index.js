import React from 'react';
import { string, bool, func, node } from 'prop-types';
import { StyledIcon } from './styled';

const propTypes = {
    color: string,
    scale: string,
    className: string,
    loading: bool,
    onClick: func,
    children: node,
};

const defaultProps = {
    color: undefined,
    scale: undefined,
    className: undefined,
    loading: undefined,
    onClick: undefined,
    children: undefined,
};

const Icon = ({ color, scale, className, loading, onClick, children }) => {
    const classname = `material-icons ${className || ''}`;
    const iconValue = loading ? 'autorenew' : children;
    return (
        <StyledIcon className={classname} color={color} scale={scale} loading={loading} onClick={onClick}>
            {iconValue}
        </StyledIcon>
    );
};

Icon.propTypes = propTypes;
Icon.defaultProps = defaultProps;

export default Icon;
