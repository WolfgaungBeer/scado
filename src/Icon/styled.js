import styled, { keyframes, css } from 'styled-components';

const rotationKeyframes = keyframes`
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
`;

export const StyledIcon = styled.i`
    cursor: ${props => props.onClick && 'pointer'};
    vertical-align: -16%;
    padding: 0 0.25rem;
    color: ${props => props.theme.color[props.color] || 'inherit'};
    font-size: ${props => props.theme.scale[props.scale] || 'inherit'};
    ${(props) => {
        if (props.loading) {
            return css`
                display: inline-block;
                animation: ${rotationKeyframes} 0.85s linear infinite;
            `;
        }
        return undefined;
    }};
`;
