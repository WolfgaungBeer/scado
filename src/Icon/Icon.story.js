import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean } from '@storybook/addon-knobs';
import Icon from './';


storiesOf('Icon', module)
    .add('default', () => (
        <Icon icon="check_circle" />
    ))
    .add('with props', () => (
        <Icon
            icon={text('icon', 'check_circle')}
            color={text('color', 'success')}
            scale={text('scale', 'l')}
            loading={boolean('loading', false)}
            onClick={action('icon clicked')}
        />
    ));
