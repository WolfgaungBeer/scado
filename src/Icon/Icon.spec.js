import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Icon from './';

it('renders the Icon Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Icon icon="check_circle" />
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Icon icon="check_circle" color="success" scale="l" />
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Icon icon="check_circle" color="success" scale="l" loading />
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
