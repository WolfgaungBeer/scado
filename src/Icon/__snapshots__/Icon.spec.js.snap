// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Icon Component correctly 1`] = `
<i
  className="material-icons  sc-ifAKCX kRKplF"
  color={undefined}
  onClick={undefined}
  scale={undefined}
>
  check_circle
</i>
`;

exports[`renders the Icon Component correctly 2`] = `
<i
  className="material-icons  sc-ifAKCX biNVRB"
  color="success"
  onClick={undefined}
  scale="l"
>
  check_circle
</i>
`;

exports[`renders the Icon Component correctly 3`] = `
<i
  className="material-icons  sc-ifAKCX kWifBW"
  color="success"
  onClick={undefined}
  scale="l"
>
  autorenew
</i>
`;
